pypdf/generic/_data_structures.py

```diff
@@ -818,6 +818,20 @@ def _clone(
             pass
         super()._clone(src, pdf_dest, force_duplicate, ignore_fields)
 
+    def get_data(self) -> bytes:
```

Comment:
```suggestion
    def get_data(self) -> bytes:
```
So far from I remember _data does not have the same meaning in an encodedStream (where _data contains the compressed data) and DecodedStream(where the data are clear). raising up the set_data into content stream will leave people think set_data on an encoded stream is valid where the results are not good (some side fields needs to be set also)

Reply:
> So far from I remember _data does not have the same meaning in an encodedStream (where _data contains the compressed data) and DecodedStream(where the data are clear)

Interesting. Is this documented somewhere?

Accessing a private property (`._data`) from outside the `ContentStream` class seemed like a code smell to me,
but if there is a semantic meaning to using `.data` instead of `get/set_data()`,
I should take care about this.

> raising up the set_data into content stream will leave people think set_data on an encoded stream is valid where the results are not good (some side fields needs to be set also)

I do not really understand: `EncodedStreamObject.set_data()` existed prior to this PR,
and has been used in several places, right?
But you mention that it should not be used / is not valid?